dev/dns.go

```diff
@@ -1,17 +1,18 @@
 package dev
 
 import (
+	"fmt"
 	"net"
-	"sync"
 	"time"
 
 	"github.com/miekg/dns"
+	"gopkg.in/tomb.v2"
 )
 
-const DefaultAddress = ":9253"
-
 type DNSResponder struct {
-	Address string
+	Address   string
+	udpServer *dns.Server
+	tcpServer *dns.Server
```

Comment:
expose pointers to the servers were running internally, so we can shut them down gracefully in tests.